pkg/controller/pxc/upgrade.go

```diff
@@ -24,6 +24,7 @@ import (
 
 func (r *ReconcilePerconaXtraDBCluster) updatePod(sfs api.StatefulApp, podSpec *api.PodSpec, cr *api.PerconaXtraDBCluster, initContainers []corev1.Container) error {
 	currentSet := sfs.StatefulSet()
+	annotations := currentSet.Spec.Template.Annotations
```

Comment:
It looks like something happens here. But in fact, it is equal to 

```go
var currentSet *appsv1.StatefulSet

annotations := make(map[string]string)
```

and later on 
```go
	for k, v := range annotations {
		currentSet.Spec.Template.Annotations[k] = v
	}
```
doesn't do any useful job either.

Moreover, since `map` is a pointer type all these maps reassigning looks frightening. Ok, `r.client.Get` during `currentSet` update  will use DeepCopy() to create a new map so `annotations ` and  `currentSet.Spec.Template.Annotations` won't be pointing to the same map. But I suggest eradicating these redundant moves.

Just leave
```go
var currentSet *appsv1.StatefulSet
```
 and no need in code in L27 & L45-49

Reply:
https://github.com/percona/percona-xtradb-cluster-operator/pull/495/commits/8e13ea090654a503cdbde00eb0ee4a83b791d608